bigtable/tests/data_integration_test.cc

```diff
@@ -30,66 +92,231 @@ int main(int argc, char* argv[]) try {
-      found = true;
-      break;
-    }
-  }
-  if (not found) {
-    throw std::runtime_error("Could not find table test table");
-  }
-  std::cout << table_name << " found via ListTables()" << std::endl;
 
-  auto client = bigtable::CreateDefaultDataClient(project_id, instance_id,
-                                                  bigtable::ClientOptions());
-  bigtable::Table table(client, table_name);
+  (void)::testing::AddGlobalTestEnvironment(
+      new DataTestEnvironment(project_id, instance_id, table_name));
 
-  // TODO(#29) we should read these rows back when we have a read path
-  auto mutation = bigtable::SingleRowMutation("row-key-0");
-  mutation.emplace_back(bigtable::SetCell(family, "col0", 0, "value-0-0"));
-  mutation.emplace_back(bigtable::SetCell(family, "col1", 0, "value-0-1"));
-  table.Apply(std::move(mutation));
-  std::cout << "row-key-0 mutated successfully" << std::endl;
-
-  mutation = bigtable::SingleRowMutation("row-key-1");
-  mutation.emplace_back(bigtable::SetCell(family, "col0", 0, "value-1-0"));
-  mutation.emplace_back(bigtable::SetCell(family, "col1", 0, "value-1-1"));
-  table.Apply(std::move(mutation));
-  std::cout << "row-key-1 mutated successfully" << std::endl;
-
-  bigtable::BulkMutation bulk{
-      bigtable::SingleRowMutation("row-key-2",
-                                  {bigtable::SetCell(family, "c0", 0, "v0"),
-                                   bigtable::SetCell(family, "c1", 0, "v1")}),
-      bigtable::SingleRowMutation("row-key-3",
-                                  {bigtable::SetCell(family, "c0", 0, "v2"),
-                                   bigtable::SetCell(family, "c1", 0, "v3")}),
-  };
-  table.BulkApply(std::move(bulk));
-  std::cout << "bulk mutation successful" << std::endl;
+  return RUN_ALL_TESTS();
 
-  return 0;
 } catch (bigtable::PermanentMutationFailure const& ex) {
   std::cerr << "bigtable::PermanentMutationFailure raised: " << ex.what()
             << " - " << ex.status().error_message() << " ["
             << ex.status().error_code()
             << "], details=" << ex.status().error_details() << std::endl;
+  int count = 0;
+  for (auto const& failure : ex.failures()) {
+    std::cerr << "failure[" << count++
+              << "] {key=" << failure.mutation().row_key() << "}" << std::endl;
+  }
   return 1;
 } catch (std::exception const& ex) {
   std::cerr << "Standard exception raised: " << ex.what() << std::endl;
   return 1;
+} catch (...) {
+  std::cerr << "Unknown exception raised." << std::endl;
+  return 1;
+}
+
+namespace {
+/**
+ * Compare two cells, think about the spaceship operator.
+ *
+ * @return `< 0` if lhs < rhs, `0` if lhs == rhs, and `> 0` otherwise.
+ */
+int CellCompare(bigtable::Cell const& lhs, bigtable::Cell const& rhs);
+
+/**
+ * Compare two sets of cells.
+ *
+ * Unordered because ReadRows does not guarantee a particular order.
+ */
+void CheckEqualUnordered(std::vector<bigtable::Cell> expected,
```

Comment:
Ditto re: refactor to `bigtable/client/testing`.

Reply:
Fixed!